<launch>

  <!-- ARGUMENTS -->
  <arg name="use_joy" default="false"           />
  <arg name="joy_dev" default="/dev/input/js0"  />

  <!--
  DRIVERS
  -->

  <!-- Cozmo driver -->
  <node ns="cozmo"
        pkg="cozmo_driver" type="cozmo_driver.py" name="cozmo_driver_node"
        output="screen" >
    <param name="camera_info_url" value="file://$(find cozmo_driver)/config/cozmo_camera.yaml" />
  </node>

  <!--
  INPUT
  -->

  <!-- Control -->
  <node pkg="twist_mux" type="twist_mux" name="twist_mux_node">
    <rosparam command="load" file="$(find cozmo_driver)/config/twist_mux.yaml" />
    <remap from="cmd_vel_out" to="cozmo/cmd_vel"/>
  </node>

  <!-- Teleoperation -->
  <node unless="$(arg use_joy)"
        pkg="cozmo_driver" type="teleop_key.py" name="teleop_node"
        output="screen" >
    <remap from="cmd_vel" to="key_teleop/cmd_vel"/>
    <remap from="head_angle"  to="cozmo/head_angle"   />
    <remap from="lift_height" to="cozmo/lift_height"  />
  </node>

  <!-- Joystick Teleoperation (optional) -->
  <group if="$(arg use_joy)"
         ns="joy_teleop">

    <rosparam command="load" file="$(find cozmo_driver)/config/teleop.yaml" />

    <node pkg="joy" type="joy_node" name="joy_node" >
      <param name="dev" value="$(arg joy_dev)" />
    </node>

    <node pkg="teleop_twist_joy" type="teleop_node" name="teleop_node" />
  </group>

  <!-- Head angle / lift height teleoperation -->
  <node if="$(arg use_joy)"
      pkg="cozmo_driver" type="head_lift_joy.py" name="head_lift_joy_node"
      output="screen" >
    <remap from="joy"         to="joy_teleop/joy"     />
    <remap from="head_angle"  to="cozmo/head_angle"   />
    <remap from="lift_height" to="cozmo/lift_height"  />
  </node>

  <!--
  VISUALIZATION
  -->
  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find cozmo_driver)/config/cozmo.rviz"
        output="screen" />

</launch>